jenkins/public_nightly.Jenkinsfile

```diff
@@ -8,466 +8,113 @@ pipeline {
+                  'ethmac asap7', 
+                  'gcd asap7', 
+                  'ibex asap7', 
+                  'jpeg asap7', 
+                  'sha3 asap7', 
+                  'uart asap7', 
+                  'aes nangate45', 
+                  'black_parrot nangate45', 
+                  'bp_be_top nangate45', 
+                  'bp_fe_top nangate45', 
+                  'bp_multi_top nangate45', 
+                  'dynamic_node nangate45', 
+                  'gcd nangate45', 
+                  'ibex nangate45', 
+                  'jpeg nangate45',
+                  'swerv nangate45', 
+                  'swerv_wrapper nangate45', 
+                  'tinyRocket nangate45', 
+                  'aes sky130hd', 
+                  'chameleon sky130hd', 
+                  'chameleon_hier sky130hd',
+                  'gcd sky130hd', 
+                  'ibex sky130hd', 
+                  'jpeg sky130hd', 
+                  'microwatt sky130hd', 
+                  'riscv32i sky130hd', 
+                  'aes sky130hs', 
+                  'gcd sky130hs', 
+                  'ibex sky130hs', 
+                  'jpeg sky130hs', 
+                  'riscv32i sky130hs';
+          }
+        }
+        stages{
+          stage("Checkout master branch") {
+            steps {
+              checkout([$class: "GitSCM",
+                    branches: [[name: "*/master"]],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [
+                      [
+                        $class: "SubmoduleOption",
+                        disableSubmodules: false,
+                        parentCredentials: true,
+                        recursiveSubmodules: true,
+                        reference: "",
+                        trackingSubmodules: false
+                      ],
+                      [
+                        $class: "RelativeTargetDirectory",
+                        relativeTargetDir: "tools/OpenROAD"
+                      ]
+                    ],
+                    submoduleCfg: [],
+                    userRemoteConfigs: [
+                      [
+                      credentialsId: "openroad-ci",
+                      url: "https://github.com/The-OpenROAD-Project/OpenROAD"
+                      ]
+                    ]
+                  ])
+            }
+          }
+          stage("Build") {
+            environment {
+              OPENROAD_FLOW_NO_GIT_INIT = 1;
+            }
+            steps {
+              sh "./build_openroad.sh --local";
+              stash name: "install", includes: "tools/install/**";
+            }
+          }
+          stage("Docker") {
+            agent any;
+            steps {
+              sh "./build_openroad.sh";
+              sh 'docker run -u $(id -u ${USER}):$(id -g ${USER}) -v $(pwd)/flow/platforms:/OpenROAD-flow-scripts/flow/platforms:ro openroad/flow-scripts flow/test/test_helper.sh';
+            }
+          }
```

Comment:
Checkout, build and docker stages should be outside of the scope of the matrix {}.